File: src/BooksLibrary.Api/Domain/Book.cs
namespace BooksLibrary.Api.Domain;

public class Book
{
    public long Id { get; set; }

    public string Title { get; set; } = string.Empty;

    public string Author { get; set; } = string.Empty;

    public string ISBN { get; set; } = string.Empty;

    public int Year { get; set; }

    public long CategoryId { get; set; }

    public bool IsAvailable { get; set; } = true;

    public DateTimeOffset Created { get; set; }

    public DateTimeOffset LastChange { get; set; }
}

File: src/BooksLibrary.WebApp/src/app/core/models/book.ts
export interface Book {
  id: number;
  title: string;
  author: string;
  isbn: string;
  year: number;
  categoryId: number;
  isAvailable: boolean;
}

export interface SaveBookRequest {
  title: string;
  author: string;
  isbn: string;
  year: number;
  categoryId: number;
}